src/Microsoft.AspNet.OData.Shared/TypeHelper.cs

```diff
@@ -574,7 +574,7 @@ internal static PropertyInfo GetProperty(Type targetType, string propertyName)
         /// <returns>true if the  type has a default constructor; otherwise returns false.</returns>
         internal static bool HasDefaultConstructor(Type type)
         {
-            return (!type.IsClass) || (!type.IsAbstract) || type.GetConstructors(BindingFlags.Public | BindingFlags.Instance)
+            return (!type.IsClass) || type.GetConstructors(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static) 
```

Comment:
We made this change with the deserialization changes. I can't remember why the change was requested but there are some tests that are failing because of that `(!type.IsAbstract)` change. so an abstract class can have a constructor even though it cannot be instantiated right?